index.js

```diff
@@ -320,13 +325,93 @@ module.exports = {
     return options;
   },
 
+  _shouldHandleTypeScript(config) {
+      let emberCLIBabelConfig = config['ember-cli-babel'] || {};
+      if (typeof emberCLIBabelConfig.enableTypeScriptTransforms === 'boolean') {
+        return emberCLIBabelConfig.enableTypeScriptTransforms;
+      }
+      let typeScriptAddon = this.parent.addons
+        && this.parent.addons.find(a => a.name === 'ember-cli-typescript');
+      return typeof typeScriptAddon !== 'undefined'
```

Comment:
This definitely works, but I _think_ we can avoid the `typeof` check. Not sure that it is or isn't better though, waht do you think?

```suggestion
      return typeScriptAddon !== undefined
```

Reply:
The `undefined` global used to be able to be redefined, so that check was dangerous, but I guess nothing in the wild allows this anymore: https://kangax.github.io/compat-table/es5/#test-Immutable_globals_undefined